src/daft-plan/src/logical_optimization/rules/push_down_limit.rs

```diff
@@ -22,11 +22,17 @@ impl PushDownLimit {
 }
 
 impl OptimizerRule for PushDownLimit {
-    fn apply_order(&self) -> ApplyOrder {
-        ApplyOrder::TopDown
+    fn try_optimize(&self, plan: Arc<LogicalPlan>) -> DaftResult<Transformed<Arc<LogicalPlan>>> {
+        plan.transform_down(|node| self.try_optimize_node(node))
     }
+}
 
-    fn try_optimize(&self, plan: Arc<LogicalPlan>) -> DaftResult<Transformed<Arc<LogicalPlan>>> {
+impl PushDownLimit {
+    #[allow(clippy::only_used_in_recursion)]
```

Comment:
why do we need this clippy allow? This should be used right?

Reply:
`&self` is not actually used other than recursive invocations. It's weird because I don't think Clippy complains if you don't use `&self` at all, only if there's a recursive call

Reply:
This is just like a quick and dirty fix to ensure that things stay the same before we start actually ripping into the rules themselves.